compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -3001,34 +2996,36 @@ static void lookupScheme3(TR::Node *node, bool unbalanced, TR::CodeGenerator *cg
       int32_t offset = TR_PPCTableOfConstants::allocateChunk(1, cg);
 
       if (offset != PTOC_FULL_INDEX)
-	 {
+         {
          offset *= TR::Compiler->om.sizeofReferenceAddress();
          TR_PPCTableOfConstants::setTOCSlot(offset, address);
          if (offset<LOWER_IMMED||offset>UPPER_IMMED)
-	    {
+            {
             generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, addrRegister, cg->getTOCBaseRegister(), cg->hiValue(offset));
             generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, addrRegister, new (cg->trHeapMemory()) TR::MemoryReference(addrRegister, LO_VALUE(offset), 8, cg));
-	    }
+            }
          else
-	    {
+            {
             generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, addrRegister, new (cg->trHeapMemory()) TR::MemoryReference(cg->getTOCBaseRegister(), offset, 8, cg));
-	    }
+            }
 
          if(isInt64)
             generateTrg1MemInstruction(cg, TR::InstOpCode::ld, node, dataRegister, new (cg->trHeapMemory()) TR::MemoryReference(addrRegister, 0, 8, cg));
          else
             generateTrg1MemInstruction(cg, TR::InstOpCode::lwz, node, dataRegister, new (cg->trHeapMemory()) TR::MemoryReference(addrRegister, 0, 4, cg));
-	 }
+         }
       else
-	 {
-         if (cg->comp()->compileRelocatableCode())
+         {
+         // the loadAddress routine here will eventually map down to TR_FixedSequence2 relo type. Might need to generate a query here that moves this JITServer
+         // check downstream.
+         if (cg->comp()->compileRelocatableCode() || cg->comp()->isOutOfProcessCompilation())
             {
-            loadAddressConstant(cg, node, address, addrRegister);
+            loadAddressConstant(cg, node, address, addrRegister, true);
```

Comment:
The thing being accessed here and below is table in the code cache that contains case values and target addresses for the `lookup` IL op, so we can add a query like `needRelocationsForLookupEvaluationData()`.

Reply:
New query available here: 50cbd5359f037696490e20cd2c7a9b2d82caf747